src/west/commands/project.py

```diff
@@ -628,7 +628,8 @@ def _fetch(project):
     # --tags is required to get tags when the remote is specified as an URL.
     if _is_sha(project.revision):
         # Don't fetch a SHA directly, as server may restrict from doing so.
-        _git(project, fetch_cmd + ' --tags -- {url}')
+        _git(project, fetch_cmd + ' --tags -- {url} '
+             'refs/heads/*:refs/west_refs/*')
```

Comment:
As discussed on Slack, I think the risk of collisions when using `west` instead of `west_refs` is low enough that the extra ergonomics of having a shorter name is worthwhile.

Reply:
Well, user is not supposed to be working on those refs.
Just as users are not supposed to work on `manifest_rev` branch.

Those are solely intended for `west` usage.
If github had allowed fetching the SHA directly, then there would be no need for fetching the whole ref-space locally.
The `refs/west_refs/*` is not linked to any git remote, and content will change with a changing url in manifest.
Thus, if user need this kind of functionality, the safest for them will be to create their own own remote, and fetch into that.

And in such case, they are probably way beyond a simple `west update` of a SHA and instead doing specific work, where they have local branches as well as a need to do `git push` to a fork.

So I don't think we should make short refs names, in order to aid the user.
Also more unique name, such as `west_refs` minimizes the (I agree, potentially small) risk of collision.

@carlescufi @nashif Thoughts ?

Reply:
I agree with @mbolivar. A remote named `west` is so unlikely  that I would rather simplify the code and also enhance the aesthetics of the refspace  for the users that decide to inspect it than ensure lower collision probability.

Reply:
ok, will chage.
But for the code, it will only change from:
```
refs/heads/*:refs/west_refs/*
```
into
```
refs/heads/*:refs/west/*
```